src/media/mod.rs

```diff
@@ -270,29 +278,30 @@ impl<'a> UploadBuilder<'a> {
         }
     }
 
+    //TODO: Second sentence needs some clarifications. As it is not clear what exactly is meant.
```

Comment:
What i mean with "by default" is that if you do not call `chunk_size`, it will use 512 KiB-sized chunks for the upload. So i guess if that phrasing didn't work for you, you could make it explicit like that:

> If this is unset, `UploadFuture` will use 512 KiB-sized chunks.

Reply:
Actually i meant this:
>This is useful for noting how often a connection is
> re-negotiated, but also for when the upload can be retried, as `UploadFuture` allows you to
> retry on failure.

Reply:
Ah, that one does come out of the blue. It's referring to the implementation, because each chunk requires a new connection, effectively. (Whether `hyper` caches these connections under the hood is worth investigating, but i treat it like it doesn't.) How about something like this:

> This is useful because `UploadFuture` allows you to retry on failure. With a smaller chunk size, Twitter can "save" the data more often. However, there's also more network overhead, since each chunk sends connection headers all over again. Larger chunk sizes are better for stable network conditions, where you can reasonably expect a large upload to succeed.

Reply:
I believe the first sentence can be omitted since it doesn't really carry much sense in context of next one.
I also simplified it:

```
With a smaller chunk size, Twitter can "save" the data more often.
However, there's also network overhead, since each chunk needs a separate HTTP request.
Larger chunk sizes are better for stable network conditions, where you can reasonably expect a large upload to succeed.
```

Reply:
I like that!